src/devdocs2zim/client.py

```diff
@@ -140,34 +139,42 @@ def sort_precedence(self) -> SortPrecedence:
         return SortPrecedence.CONTENT
 
 
-class NavigationSection(BaseModel):
-    """Represents a single section of a devdocs navigation tree."""
+class NavbarPageEntry(BaseModel):
+    """Model of the a page in the navbar."""
 
-    # Heading information for the group of links.
+    # Display name of the page e.g. "Introduction"
     name: str
-    # Links to display in the section.
-    links: list[DevdocsIndexEntry]
 
-    @computed_field
-    @property
-    def count(self) -> int:
-        """Number of links in the section."""
-        return len(self.links)
+    # Link to the page
+    href: str
+
 
-    @cached_property
-    def _contained_pages(self) -> set[str]:
-        return {link.path_without_fragment for link in self.links}
+class NavbarSectionEntry(BaseModel):
+    """Model of the a section in the navbar."""
 
-    def opens_for_page(self, page_path: str) -> bool:
-        """Returns whether this section should be rendered open for the given page."""
+    # Display name of the section e.g. "Tutorials"
+    name: str
 
-        # Some docs like Lua or CoffeeScript have all of their content in the index.
-        # Others like RequireJS are split between index and additional pages.
-        # We don't want sections opening when the user navigates to the index.
-        if page_path == "index":
-            return False
+    # Pages in the section
+    children: list[NavbarPageEntry]
 
-        return page_path in self._contained_pages
+
+class NavbarDocument(BaseModel):
+    """Model of the document to populate each page's navbar with."""
+
+    # Display name of the document e.g. "Lua"
+    name: str
+    # Link to the main root of the document. DevDocs makes this "index"
+    # implicitly.
+    landing_href: str = Field(
+        serialization_alias="landingHref", default=DEVDOCS_LANDING_PAGE
```

Comment:
Instead of handling "camelization" manually, you might be interested to add pyhump and configure a custom BaseModel which will do that for you (less risk of forgetting one at some point). See e.g. https://github.com/openzim/libretexts/blob/main/scraper/src/libretexts2zim/ui.py ; more a comment for you than something needed be done.
Note that you still have to not forget to set `by_alias=True`.

Reply:
Done, it turns out Pydantic has these built in so we don't need another external package.